blockchain/binance/client.go

```diff
@@ -59,26 +50,22 @@ func (c Client) FetchTransactionsInBlock(blockNumber int64) (TransactionsInBlock
 	}
 	var result TransactionsInBlockResponse
 	if err := resp.ToJSON(&result); err != nil {
-		log.Error("URL: " + resp.Request().URL.String())
-		log.Error("Status code: " + resp.Response().Status)
-		return TransactionsInBlockResponse{}, err
+		return TransactionsInBlockResponse{}, errors.Wrap(err, "URL: "+resp.Request().URL.String()+"; Code: "+resp.Response().Status)
 	}
 	return result, nil
 }
 
-func (c Client) FetchTransactionsByAddressAndTokenID(address, tokenID string) ([]Tx, error) {
+func (c Client) FetchTransactionsByAddressAndTokenID(address, tokenID string, txPerPage int) ([]Tx, error) {
 	startTime := strconv.Itoa(int(time.Now().AddDate(0, -3, 0).Unix() * 1000))
-	limit := strconv.Itoa(c.txPerPage)
+	limit := strconv.Itoa(txPerPage)
 	params := url.Values{"address": {address}, "txAsset": {tokenID}, "startTime": {startTime}, "limit": {limit}}
 	resp, err := c.Get("/api/v1/transactions", params)
 	if err != nil {
 		return nil, err
 	}
 	var result TransactionsInBlockResponse
 	if err := resp.ToJSON(&result); err != nil {
-		log.Error("URL: " + resp.Request().URL.String())
-		log.Error("Status code: " + resp.Response().Status)
-		return nil, err
+		return nil, errors.Wrap(err, "URL: "+resp.Request().URL.String()+"; Code: "+resp.Response().Status)
```

Comment:
Do you think it would make sense to have 1 function for call error wrappers? not just binance dex client but for the rest

Reply:
It correct for libraries. You should return details error. I add an additional message to the general error. General error need for correct stack trace.

BUT!
No need to add the wrap function to wrap!

Reply:
You will have tons of duplications, because we would need to create consistent error handling for each blockchain client

Reply:
You might need to think of better ways to handle errors